server/userModel/userData.js

```diff
@@ -0,0 +1,59 @@
+const User = require("./user");
+const bcrypt = require("bcrypt");
+const saltRounds = 10;
+const jwt = require("jsonwebtoken");
+const config = require("../config/auth.config");
+
+function encryptedPasswordWithSalt(plaintextPassword){
+    return bcrypt.hash(plaintextPassword, saltRounds);
+}
+
+function isPasswordValid(plaintextPassword,hash){
+    return bcrypt.compare(plaintextPassword, hash);
+}
+
+function createNewUserObject(user){
+    const newUser = new User({
+        email:user.email,
+        companyName:user.companyName,
+        password: encryptedPasswordWithSalt(user.password)
+    });
+
+    return newUser;
+}
+
+function saveDataToUserModel(newUser){
```

Comment:
Most of these helpers aren't really required. You can handle the logic within the routes themselves.